moxy-compiler/src/main/java/moxy/compiler/viewstate/ViewInterfaceProcessor.kt

```diff
@@ -236,13 +237,42 @@ class ViewInterfaceProcessor(
         return modifiers.contains(Modifier.STATIC)
     }
 
-    private inline fun <T, R> List<T>.flatMapToSet(transform: (T) -> Iterable<R>): Set<R> {
-        return flatMapTo(LinkedHashSet(), transform)
-    }
-
     companion object {
         private const val OPTION_DEFAULT_STRATEGY = MvpCompiler.DEFAULT_MOXY_STRATEGY
         private val DEFAULT_STATE_STRATEGY: TypeElement =
             elementUtils.getTypeElement(AddToEndSingleStrategy::class.java.canonicalName)
     }
+
+    class StockViewMethod(
+            val targetInterfaceElement: DeclaredType,
+            val element: ExecutableElement,
+            val strategy: TypeElement?,
+            val tag: String
+    )
+
+    private fun StockViewMethod.toViewMethod(): ViewMethod {
+        if(strategy == null) throw IllegalStateException("Strategy can't be null")
+        return ViewMethod(targetInterfaceElement, element, strategy, tag)
+    }
+
+    private fun TypeElement.isNotMvpViewExtend(): Boolean {
+        return this.interfaces.find { parent ->
+            if ((parent as DeclaredType).toString() == MvpView::class.java.canonicalName) true
+            else !parent.asTypeElement().isNotMvpViewExtend()
+        } == null
+    }
```

Comment:
- I think method should be named like `notExtendsMvpView`, to be more clear
 - `if` inside lambda can be replaced with simple expression `if (something()) true else other()` -> `something() || other()`
 - `interfaces.find { condition(it) } == null` can be replaced with `interfaces.all { !condition() }`
 - `(parent as DeclaredType)` is unnecessary

Applying all those changes, it would look like this:

```
private fun TypeElement.notExtendsMvpView(): Boolean {
    return interfaces.all { parent ->
        parent.toString() != MvpView::class.java.canonicalName && parent.asTypeElement().notExtendsMvpView()
    }
}
```